spacy/training/example.pyx

```diff
@@ -153,24 +153,74 @@ cdef class Example:
     def get_aligned(self, field, as_string=False):
         """Return an aligned array for a token attribute."""
         align = self.alignment.x2y
-
-        vocab = self.reference.vocab
         gold_values = self.reference.to_array([field])
-        output = [None] * len(self.predicted)
-        for token in self.predicted:
-            values = gold_values[align[token.i]]
-            values = values.ravel()
-            if len(values) == 0:
-                output[token.i] = None
-            elif len(values) == 1:
-                output[token.i] = values[0]
-            elif len(set(list(values))) == 1:
+
+        def _vectorized():
```

Comment:
I'd personally prefer making these into separate private functions, as that helps with readability.